sml/kmeans/kmeans_test.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2023 Ant Group Co., Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import json
+import jax.numpy as jnp
+
+# from sklearn.metrics import roc_auc_score, explained_variance_score
+import spu.utils.simulation as spsim
+import spu.spu_pb2 as spu_pb2  # type: ignore
+
+# TODO: unify this.
+import examples.python.utils.dataset_utils as dsutil
+
+from sml.kmeans.kmeans import KMEANS
+
+
+class UnitTests(unittest.TestCase):
+    def test_kmeans(self):
+        sim = spsim.Simulator.simple(
+            3, spu_pb2.ProtocolKind.ABY3, spu_pb2.FieldType.FM64
+        )
+
+        def proc(x1, x2):
+            x = jnp.concatenate((x1, x2), axis=1)
+            model = KMEANS(
+                k=2,
+                data_len=x.shape[0],
+                max_iter=10
+            )
+
+            return model.fit(x).predict(x)
+        
+        DATASET_CONFIG_FILE = "examples/python/conf/ds_breast_cancer_basic.json"
```

Comment:
Can we add some common used test dataset for KMeans? And also verify the result to meet our expectation.

Reply:
Different tasks need different datasets. For clustering task, `sklearn.datasets.make_blobs` can generate some good dummy dataset which can be used for verification.